### PR TITLE
process: expose AW::poll_write_vectored on unix

### PR DESCRIPTION
**Pull request by koivunej**
## Motivation

For inter-process communication over stdin and stdout it would be helpful to use vectored writes.

## Solution

Expose already existing vectored facilities in `tokio::io::PollEvented` on the `AsyncWrite` implementation of `tokio::process::ChildStdin` by providing a forwarding implementation instead of using the default non-vectoring implementation.

Adjusts the `tokio::io::PollEvented` vectored writes support to be enabled on `feature = "process"`.

### CODE REVIEW

**Comment by koivunej** [comment]
Quite unsure how to test this. `tests-integration` has `process-stdio.rs`, one could make a test that a `cat` being fed on `cfg!(unix)` a poll_write_vectored completes with 1 poll?

**Review by Darksonn — APPROVED** [review]
Looks good to me.

File: tests-integration/tests/process_stdio.rs [hunk 1]
```diff
@@ -190,3 +190,54 @@ async fn pipe_from_one_command_to_another() {
...
+    let (write, read, status) = future::join3(write, read, cat.wait()).await;
+
+    assert!(status.unwrap().success());
+
+    let writes_completed = write.unwrap();
+    // on unix our small payload should always fit in whatever default sized pipe with a single
+    // syscall. if multiple are used, then the forwarding does not work, or we are on a platform
+    // for which the `std` does not support vectored writes.
+    #[cfg(target_family = "unix")]
+    assert_eq!(writes_completed, 1);
```
Comment (Darksonn) [inline tests-integration/tests/process_stdio.rs:240]: How about doing it like this?
```suggestion
    assert_eq!(writes_completed == 1, stdin.is_write_vectored());
```

Comment (koivunej) [reply tests-integration/tests/process_stdio.rs]: Small change needed, will fix soon.

Comment (koivunej) [reply tests-integration/tests/process_stdio.rs]: Fixed in force-pushed [fca1ac6](https://github.com/tokio-rs/tokio/pull/5216/commits/fca1ac63ed84a3a0581206ccacae621a4fff7afb)

**Review by Darksonn — APPROVED** [review]
Ok looks good to me.

**Comment by koivunej** [comment]
Post merge CI failure seems unrelated: https://github.com/tokio-rs/tokio/actions/runs/3515029698/jobs/5889784158#step:4:1200

**Comment by Darksonn** [comment]
Yes, that seems to be due to a flaky test. I believe it is related to #5207.

**Pull request merged**